### PR TITLE
refactor oracle contracts

### PR DESCRIPTION
**Pull request by yuval-weiss**

### CODE REVIEW

**Comment by mijovic** [comment]
Looks good to me, maybe @StefanVitor can take a look as well.

File: test/ChainlinkPriceOracle.t.sol [hunk 1]
```diff
@@ -4,17 +4,17 @@ pragma solidity 0.8.19;
 import { Test } from "forge-std/Test.sol";
 import { AggregatorV3Interface } from "@smartcontractkit/chainlink/interfaces/AggregatorV3Interface.sol";
 import { IPriceOracle } from "../contracts/Oracles/Interfaces/IPriceOracle.sol";
-import { ChainlinkPriceOracleWETH } from "../contracts/Oracles/ChainlinkPriceOracleWETH.sol";
+import { ChainlinkPriceOracle } from "../contracts/Oracles/ChainlinkPriceOracle.sol";
 
-contract ChainlinkPriceOracleWETHTest is Test {
+contract ChainlinkPriceOracleTest is Test {
```
Comment (StefanVitor) [inline test/ChainlinkPriceOracle.t.sol:9]: I wouldn't rename this, how and for other coins, we could add tests to see is setup good.

File: test/TellorPriceOracle.t.sol [hunk 2]
```diff
@@ -4,17 +4,17 @@ pragma solidity 0.8.19;
 import { Test } from "forge-std/Test.sol";
 import { ITellor } from "../contracts/Dependencies/ITellor.sol";
 import { IPriceOracle } from "../contracts/Oracles/Interfaces/IPriceOracle.sol";
-import { TellorPriceOracleWETH } from "../contracts/Oracles/TellorPriceOracleWETH.sol";
+import { TellorPriceOracle } from "../contracts/Oracles/TellorPriceOracle.sol";
 
-contract TellorPriceOracleWETHTest is Test {
+contract TellorPriceOracleTest is Test {
```
Comment (StefanVitor) [inline test/TellorPriceOracle.t.sol:9]: Same as for Chainlink ("I wouldn't rename this, how and for other coins, we could add tests to see is setup good.")

File: contracts/Oracles/ChainlinkPriceOracleWstETH.sol [hunk 3]
```diff
@@ -1,29 +1,34 @@
...
 
-contract ChainlinkPriceOracleWstETH is BaseChainlinkPriceOracle, BasePriceOracleWstETH {
-    // --- Constants ---
+contract ChainlinkPriceOracleWstETH is IPriceOracleWstETH, ChainlinkPriceOracle {
+    using Fixed256x18 for uint256;
 
-    uint256 public constant override DEVIATION = 1e16; // 1%
+    // --- Immutables ---
+
+    IWstETH public immutable override wstETH;
```
Comment (StefanVitor) [inline contracts/Oracles/ChainlinkPriceOracleWstETH.sol:14]: This part is now duplicated, and earlier, it wasn't (there was an abstract class for this)

File: contracts/Oracles/ChainlinkPriceOracleWstETH.sol [hunk 4]
```diff
@@ -1,29 +1,34 @@
...
     )
-        BaseChainlinkPriceOracle(_priceAggregatorAddress)
-        BasePriceOracleWstETH(_wstETH)
-    // solhint-disable-next-line no-empty-blocks
-    { }
-
-    // --- Functions ---
+        ChainlinkPriceOracle(priceAggregatorAddress_, deviation_)
+    {
+        if (address(wstETH_) == address(0)) {
```
Comment (StefanVitor) [inline contracts/Oracles/ChainlinkPriceOracleWstETH.sol:25]: Same as previous ("this part is now duplicated, and earlier, it wasn't (there was abstract class for this) ")

File: contracts/Oracles/ChainlinkPriceOracleWstETH.sol [hunk 5]
```diff
@@ -1,29 +1,34 @@
...
+            revert InvalidWstETHAddress();
+        }
+        wstETH = IWstETH(wstETH_);
+    }
 
     function _formatPrice(uint256 price, uint256 answerDigits) internal view override returns (uint256) {
-        return _convertIntoWstETHPrice(price, answerDigits);
+        return super._formatPrice(price, answerDigits).mulDown(wstETH.stEthPerToken());
```
Comment (StefanVitor) [inline contracts/Oracles/ChainlinkPriceOracleWstETH.sol:32]: Same as previous ("this part is now duplicated, and earlier, it wasn't (there was abstract class for this) ")

File: contracts/Oracles/Interfaces/IPriceOracle.sol [hunk 6]
```diff
@@ -4,6 +4,11 @@ pragma solidity 0.8.19;
 import { IWstETH } from "../../Dependencies/IWstETH.sol";
```
Comment (StefanVitor) [inline contracts/Oracles/Interfaces/IPriceOracle.sol:4]: This could be deleted

File: contracts/Oracles/ChainlinkPriceOracle.sol [hunk 7]
```diff
@@ -6,24 +6,29 @@ import { Fixed256x18 } from "@tempusfinance/tempus-utils/contracts/math/Fixed256
...
-abstract contract BaseChainlinkPriceOracle is BasePriceOracle, IChainlinkPriceOracle {
+contract ChainlinkPriceOracle is BasePriceOracle, IChainlinkPriceOracle {
     // --- Types ---
 
     using Fixed256x18 for uint256;
 
     // --- Constants & immutables ---
 
     AggregatorV3Interface public immutable override priceAggregator;
+    uint256 public immutable override DEVIATION;
```
Comment (StefanVitor) [inline contracts/Oracles/ChainlinkPriceOracle.sol:17]:
```suggestion
    uint256 public immutable override deviation;
```

Comment (yuval-weiss) [reply contracts/Oracles/ChainlinkPriceOracle.sol:17]: we can't because `PriceFeed` already depends on `DEVIATION` and this would break the interface

Comment (StefanVitor) [reply contracts/Oracles/ChainlinkPriceOracle.sol:17]: Yeah, totally forgot about that.

File: test/ChainlinkPriceOracleWETH.t.sol [hunk 8]
```diff
@@ -4,17 +4,17 @@ pragma solidity 0.8.19;
...
     AggregatorV3Interface public constant aggregatorV3ETH =
         AggregatorV3Interface(0x5f4eC3Df9cbd43714FE2740f5E3616155c5b8419);
-    ChainlinkPriceOracleWETH public chainlinkPriceOracleWETH;
+    ChainlinkPriceOracle public chainlinkPriceOracleWETH;
 
     function setUp() public {
         vm.createSelectFork("mainnet", 17_484_072);
 
-        chainlinkPriceOracleWETH = new ChainlinkPriceOracleWETH(aggregatorV3ETH);
+        chainlinkPriceOracleWETH = new ChainlinkPriceOracle(aggregatorV3ETH, 1e15);
```
Comment (StefanVitor) [inline test/ChainlinkPriceOracleWETH.t.sol:17]:
```suggestion
        chainlinkPriceOracleWETH = new ChainlinkPriceOracle(aggregatorV3ETH, 5e15);
```

Comment (StefanVitor) [reply test/ChainlinkPriceOracleWETH.t.sol:17]: Wen somebody deploys a contract, to don't make a mistake if check here.

File: test/TellorPriceOracleWETH.t.sol [hunk 9]
```diff
@@ -4,17 +4,17 @@ pragma solidity 0.8.19;
...
     ITellor public constant tellorOracle = ITellor(0xD9157453E2668B2fc45b7A803D3FEF3642430cC0);
-    TellorPriceOracleWETH public tellorPriceOracleWETH;
+    TellorPriceOracle public tellorPriceOracleWETH;
 
     function setUp() public {
         vm.createSelectFork("mainnet", 17_484_688);
 
         tellorPriceOracleWETH =
-            new TellorPriceOracleWETH(tellorOracle, keccak256(abi.encode("SpotPrice", abi.encode("eth", "usd"))));
+            new TellorPriceOracle(tellorOracle, keccak256(abi.encode("SpotPrice", abi.encode("eth", "usd"))), 5e15);
```
Comment (StefanVitor) [inline test/TellorPriceOracleWETH.t.sol:17]:
```suggestion
            new TellorPriceOracle(tellorOracle, keccak256(abi.encode("SpotPrice", abi.encode("eth", "usd"))), 0);
```

File: test/TellorPriceOracleWstETH.t.sol [hunk 10]
```diff
@@ -17,7 +17,7 @@ contract TellorPriceOracleWstETHTest is Test {
 
         tellorPriceOracleWstETH = new TellorPriceOracleWstETH(
             tellorOracle, keccak256(abi.encode("SpotPrice", abi.encode("steth", "usd"))
-        ), wstETH);
+        ), wstETH, 5e15);
```
Comment (StefanVitor) [inline test/TellorPriceOracleWstETH.t.sol:20]:
```suggestion
        ), wstETH, 0);
```

File: test/ChainlinkPriceOracleWETH.t.sol (hunk 8 above)
Comment (yuval-weiss) [reply test/ChainlinkPriceOracleWETH.t.sol:17]: whoops

File: test/TellorPriceOracleWETH.t.sol (hunk 9 above)
Comment (yuval-weiss) [reply test/TellorPriceOracleWETH.t.sol:17]: currently the Tellor deviation constant is set to 5e15 though

Comment (StefanVitor) [reply test/TellorPriceOracleWETH.t.sol:17]: Yeah, in reality, it is not a problem because it is still less than Chainlink, but, in real, it is 0.

**Review by StefanVitor — APPROVED** [review]

**Pull request merged**